module/zfs/zio.c

```diff
@@ -3098,10 +3125,11 @@ zio_write_gang_member_ready(zio_t *zio)
 	VERIFY3U(BP_GET_NDVAS(zio->io_bp), <=, BP_GET_NDVAS(pio->io_bp));
 
 	mutex_enter(&pio->io_lock);
-	for (int d = 0; d < BP_GET_NDVAS(zio->io_bp); d++) {
+	for (int d = 0; d < BP_GET_NDVAS(pio->io_bp); d++) {
 		ASSERT(DVA_GET_GANG(&pdva[d]));
 		asize = DVA_GET_ASIZE(&pdva[d]);
-		asize += DVA_GET_ASIZE(&cdva[d]);
+		for (int cd = 0; cd < BP_GET_NDVAS(zio->io_bp); cd++)
+			asize += DVA_GET_ASIZE(&cdva[cd]);
```

Comment:
I vaguely remember where we ended up discussing this before, but I have feeling that here you are duplicating the ASIZE by number of children, since each child will describe exactly the same grad-children blocks.  I think this chunk should be removed, restoring the code as it was.  It should be OK after all assertions above and recent `zp_gang_copies` change we made.